crates/router/src/db/reverse_lookup.rs

```diff
@@ -21,150 +32,81 @@ pub trait ReverseLookupInterface {
     ) -> CustomResult<ReverseLookup, errors::StorageError>;
 }
 
-#[cfg(not(feature = "kv_store"))]
-mod storage {
-    use error_stack::IntoReport;
-    use router_env::{instrument, tracing};
+#[async_trait::async_trait]
+impl ReverseLookupInterface for Store {
+    #[instrument(skip_all)]
+    async fn insert_reverse_lookup(
+        &self,
+        new: ReverseLookupNew,
+        storage_scheme: enums::MerchantStorageScheme,
+    ) -> CustomResult<ReverseLookup, errors::StorageError> {
+        let redis_conn = self
+            .get_redis_conn()
+            .map_err(|err| errors::StorageError::RedisError(err))?;
+
+        let created_rev_lookup = ReverseLookup {
+            lookup_id: new.lookup_id.clone(),
+            sk_id: new.sk_id.clone(),
+            pk_id: new.pk_id.clone(),
+            source: new.source.clone(),
+            updated_by: storage_scheme.to_string(),
+        };
+
+        let ttl = self.ttl_for_kv.saturating_add(120);
 
-    use super::{ReverseLookupInterface, Store};
-    use crate::{
-        connection,
-        errors::{self, CustomResult},
-        types::storage::{
-            enums,
-            reverse_lookup::{ReverseLookup, ReverseLookupNew},
-        },
-    };
+        created_rev_lookup
```

Comment:
why are we maintaining unique constraints for reverse lookup, we arent going to drain this in db right

Reply:
> why are we maintaining unique constraints for reverse lookup, we arent going to drain this in db right

I kept it for the case when we are inserting a data that already exists, Since the first case happens for reverse lookup it will insert it in reverse lookup and throw error from the source table. Since the reverse lookup data is with a ttl this could be okay. But it will also add extra latency of inserting reverse lookup for no reason.